sdk/identity/azure-identity/src/main/java/com/azure/identity/ChainedTokenCredential.java

```diff
@@ -40,27 +41,32 @@
     public Mono<AccessToken> getToken(TokenRequestContext request) {
         List<CredentialUnavailableException> exceptions = new ArrayList<>(4);
         return Flux.fromIterable(credentials)
-                   .flatMap(p -> p.getToken(request).onErrorResume(Exception.class, t -> {
-                       if (!t.getClass().getSimpleName().equals("CredentialUnavailableException")) {
-                           return Mono.error(new ClientAuthenticationException(
+            .flatMap(p -> p.getToken(request)
+                .doOnNext(t -> logger.info("Azure Identity => Attempted credential {} returns a token",
```

Comment:
should this be doOnSuccess too ?

Reply:
Yeah I can make that change. Though I'm not exactly sure of their difference in our use cases

Reply:
Per offline discussion, `doOnNext` should be used everywhere for now, as we cannot guarantee a Mono will emit a token when `doOnSuccess` is signaled. An upstream Mono *may* terminate without emitting either a token or an error. `doOnNext()` is the more accurate mapping to the message conveyed "xxx returns a token".

Ideally a Mono from the `IdentityClient` should always return an item or an error. This may or may not be possible. The work is out of scope for this PR thus deferred for further investigation.